### PR TITLE
Fix startAfter/endBefore for orderByKeys queries

### PR DESCRIPTION
**Pull request by IanWyszynski**
We should not be passing a fallback key to startAfter if we're querying on a key index. The indexValue is the key in this case.

### CODE REVIEW

File: packages/database/src/api/Query.ts [hunk 1]
```diff
@@ -101,21 +101,15 @@ export class Query {
         'or equalTo() must be a string.';
       if (params.hasStart()) {
         const startName = params.getIndexStartName();
-        if (
```
Comment (IanWyszynski) [inline packages/database/src/api/Query.ts:104]: These just revert a previous change.

File: packages/database/src/core/view/QueryParams.ts [hunk 2]
```diff
@@ -289,13 +289,21 @@ export class QueryParams {
   }
 
   startAfter(indexValue: unknown, key?: string | null): QueryParams {
-    let childKey: string;
-    if (key == null) {
-      childKey = MAX_NAME;
+    let params: QueryParams;
+    if (this.index_ === KEY_INDEX) {
+      if (typeof indexValue === 'string') {
```
Comment (IanWyszynski) [inline packages/database/src/core/view/QueryParams.ts:294]: This check makes it so that startAfter still throws exceptions for incorrectly typed keys.

Comment (schmidt-sebastian) [reply packages/database/src/core/view/QueryParams.ts:294]: Where does it throw? Should we throw at the callsite to preserve the stacktrace of the original mistake?

Comment (IanWyszynski) [reply packages/database/src/core/view/QueryParams.ts:294]: It throws in validateQueryEndpoints_: https://github.com/firebase/firebase-js-sdk/blob/master/packages/database/src/api/Query.ts#L110, which is the same place that startAt() and endAt() throw for this case. 

I think that the way it's done now the original mistake, passing a non-string key to `startAfter()` should still be clear. I'll also update the error message there.

**Review by schmidt-sebastian — APPROVED** [review]
Please add change log entry.

**Pull request merged**